src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -332,6 +337,49 @@ public class ReleaseService : IReleaseService
                 });
         }
 
+        public async Task<Either<ActionResult, Unit>> UpdateReleasePublished(Guid releaseId,
+            ReleasePublishedUpdateRequest request)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Release>(releaseId,
+                    queryable => queryable.Include(r => r.Publication))
+                .OnSuccessDo(_userService.CheckIsBauUser)
+                .OnSuccess<ActionResult, Release, Unit>(async release =>
+                {
+                    if (release.Published == null)
+                    {
+                        return ValidationActionResult(ReleaseNotPublished);
+                    }
+
+                    // Prevent assigning a future date since it would have the effect of un-publishing the release
+                    if (request.Published > DateTime.UtcNow)
+                    {
+                        return ValidationActionResult(ReleasePublishedCannotBeFutureDate);
+                    }
+
+                    _context.Releases.Update(release);
+                    release.Published = request.Published ?? DateTime.UtcNow;
```

Comment:
We know that `request.Published` is not null by this point, so we should be able to skip this check.

Reply:
I've made a change now above to begin with this:
`var newPublishedDate = request.Published?.ToUniversalTime() ?? DateTime.UtcNow;`

and then later on here `release.Published = newPublishedDate`.

`request.Published` is optional because that's the only way I could get the `[Required]` validation enforced. If it's not marked as optional and not present in the request body, `Published` will be `0000-00-00`, i.e. the default value of `DateTime` which we don't want to happen.